squidDrone/src/propulsion/esc.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef SRC_PROPULSION_ESC_HPP_
+#define SRC_PROPULSION_ESC_HPP_
+
+#include "stm32g4xx.h"
+#include "error_types.hpp"
+
+namespace propulsion{
+
+  ///
+  /// @class Esc 
+  /// @brief The abstract Esc provides a interface for concrete Esc's to implement
+  /// The Esc abstract interface dictates the methods which all concrete Esc's have to
+  /// implement. 
+  ///
+  class Esc{
+    public:
+      ///
+      /// @brief  Default constructor may not be used, so
+      ///         it is deleted.
+      ///
+      Esc() = delete;
+
+      ///
+      /// @brief  The custom constructor is
+      ///         the one to be used. 
+      /// @param  timer The timer handle struct defined by hal driver 
+      ///               stm32g4xx_hal_tim.h. 
+      /// @param  channel The used channel for this timer
+      /// The only thing this constructor does, is to set the protected 
```

Comment:
/// The only thing this constructor does, is to set the protected 
      /// member variable timer_. <---- and protected member variable channel_

Reply:
fixed in [this commit](https://github.com/elheck/Squiddrone/pull/168/commits/04989db9ed3460ed454f9233b555d70d0787455c)